src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -55,122 +61,172 @@ public static SyncerFactory of(
+              buildSyncer(syncOptionValue, runnerNameOptionValue, isSyncProjectSyncCustomObjects);
+      syncersToRunParallel.add(
+          syncer.sync(runnerNameOptionValue, isFullSync).toCompletableFuture());
+    }
+    return CompletableFuture.allOf(syncersToRunParallel.toArray(new CompletableFuture[0]));
+  }
+
+  @Nonnull
+  private static List<SyncModuleOption> validateAndCollectSyncOptionValues(
+      @Nonnull final String[] syncOptionValues) {
 
-    if (isBlank(syncOptionValue)) {
+    if (null == syncOptionValues) {
       final String errorMessage =
           format(
-              "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
-              SYNC_MODULE_OPTION_SHORT, SYNC_MODULE_OPTION_LONG, SYNC_MODULE_OPTION_DESCRIPTION);
+              "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+              null,
+              SYNC_MODULE_OPTION_SHORT,
+              SYNC_MODULE_OPTION_LONG,
+              SYNC_MODULE_OPTION_DESCRIPTION);
+      throw new IllegalArgumentException(errorMessage);
+    }
+    if (syncOptionValues.length == 1 && SYNC_MODULE_OPTION_ALL.equals(syncOptionValues[0])) {
+      return Arrays.asList(SyncModuleOption.values());
+    } else {
+      return Arrays.stream(syncOptionValues)
+          .map(
+              syncOption -> {
+                if (isBlank(syncOption)) {
+                  final String errorMessage =
+                      format(
+                          "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
 
-      return exceptionallyCompletedFuture(new IllegalArgumentException(errorMessage));
+                  throw new IllegalArgumentException(errorMessage);
+                }
+
+                final String trimmedValue = syncOption.trim();
+                try {
+                  return SyncModuleOption.getSyncModuleOptionBySyncOptionValue(trimmedValue);
+                } catch (IllegalArgumentException e) {
+                  final String errorMessage =
+                      format(
+                          "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+                          trimmedValue,
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
+                  throw new IllegalArgumentException(errorMessage);
+                }
+              })
+          .collect(Collectors.toList());
     }
+  }
 
-    Syncer<
-            ? extends ResourceView,
-            ?,
-            ? extends BaseSyncStatistics,
-            ? extends BaseSyncOptions<?, ?>,
-            ? extends QueryDsl<?, ?>,
-            ? extends BaseSync<?, ?, ?>>
-        syncer;
+  private static Collection<List<SyncModuleOption>> groupSyncModuleOptions(
```

Comment:
What about documenting this algorithm as Javadoc? So anyone who is not familiar with the algorithm could edit it when it needs to be changed in future?

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/219/commits/39701b0756b605aa70a49b931d101e18bacc033f